doc/developer/design/20240502_deprecate_subsources.md

```diff
@@ -0,0 +1,228 @@
+`CREATE TABLE`.
+
+(h/t @benesch for this)
+
+---
+
+Create a Kafka source that reads from a topic named `t`:
+
+```sql
+CREATE SOURCE kafka_src
+FROM KAFKA CONNECTION kafka_conn (TOPIC 't')
+FOR ALL TABLES
+```
+
+`kafka_src` is the progress relation for the source, and a table named `t` is
+automatically created in the same schema as `kafka_src`.
+
+Note that we could also probably make `FOR ALL TABLES` implied for Kafka
+sources.
+
+---
+
+Create a Kafka source that reads from a topic named `t` with a custom table
+name:
+
+```sql
+CREATE SOURCE kafka_src
+FROM KAFKA CONNECTION kafka_conn (TOPIC 't')
+FOR TABLES (t AS my_t)
+```
+
+`kafka_src` is again the progress relation for the source. The data is in a
+table named `my_t`. You could also imagine more complex scenarios where you can
+output the same topic with multiple schemas. What that ends up looking like is
+under discussion.
+
+---
+
+Create a PostgreSQL source:
+
+```sql
+CREATE SOURCE postgres_src
+FROM POSTGRES CONNECTION postgres_conn (PUBLICATION p)
+FOR ALL TABLES
+```
+
+`postgres_src` is the progress relation; there is no separate
+`postgres_src_progress` relation created. Otherwise this command works the same
+way that it does today.
+
+---
+
+Add a new table to a PostgreSQL source:
+
+```sql
+CREATE TABLE my_new_table FROM SOURCE postgres_src (upstream_schema.upstream_table);
+```
+
+This would replace the `ALTER SOURCE ... ADD {SUBSOURCE|TABLE}` command.
+
+---
+
+Add a new table to a Kafka source:
+
+```sql
+CREATE TABLE kafka_copy FROM SOURCE kafka_src ("t");
+```
+
+At first this looks weird. Why would you ever want two copies of a Kafka source?
+It turns out this lets you incorporate upstream schema changes without losing
+reclocking information! This would get us reusable reclocking
+(https://github.com/MaterializeInc/materialize/issues/17021), without needing to
+introduce a new `SOURCE PROGRESS` concept.
+
+Note that support for this is blocked on at least
+https://github.com/MaterializeInc/materialize/issues/24843, and also additional
+work to support multi-output sources in Kafka https://github.com/MaterializeInc/materialize/issues/26994.
+
+#### System tables
+
+-   `mz_tables` should grow an `managing_source_id` column, whose type is a
+    nullable string (the value being the source's `GlobalId` if it exists).
+
+-   The output of `SHOW TABLES` should include a new column called `managed_by`
+    returns the name of the source that this table is managed by using whatever
+    level of qualification we use in `SHOW SUBSOURCES` today. It will be `NULL`
+    for all "regular" tables.
+
+-   `SHOW SUBSOURCES` gets deprecated.
+
+-   `SHOW TABLES` gets a new clause akin to `ON`, which is `MANAGED BY` and lets
+    you see which tables a given source manages.
+
+#### TEXT + IGNORE COLUMNS
+
+These options can remain on `CREATE SOURCE` statements, and the `ADD SUBSOURCE`
+variants of them will move to instead be `CREATE TABLE` variants. We could also
```

Comment:
Can you describe what this would look like for `CREATE TABLE`? Would it just be 
```
CREATE TABLE my_new_table FROM SOURCE postgres_src (upstream_schema.upstream_table) WITH (TEXT COLUMNS (upstream_table.column_1), IGNORE COLUMNS (upstream_table.column_2))
```
?

Reply:
Yes. They will be the same as the `ADD SUBSOURCE` options but as options on `CREATE TABLE`. This is what I meant by move.